app/models/ebook.rb

```diff
@@ -43,13 +46,29 @@ class Ebook < ApplicationRecord
       presence: true
 
   validates :isbn,
-    presence: true
+    presence: true,
+    format: {
+      with: /\A978-\d{10}\z/,
+      message: "has not a valid format"
+    }
+
+  validate :publication_date_cannot_be_in_the_future
 
   scope :live, -> { where(status: :live) }
   scope :filter_by_tags, ->(tag_ids) { joins(:ebook_tags).where(ebook_tags: { tag_id: tag_ids }) }
   scope :filter_by_users, ->(user_ids) { where(user_id: user_ids) }
 
   def discount_value(discount)
-    (price * (discount / 100.0)).round(2)
+    if discount >= 0 && discount <= 100
```

Comment:
Isto não deveria de ser `discount > 0 && discount <= 100`? Se o desconto for 0, penso que não queiras executar isto. Já agora, o Rails dá-te um método chamado `in?` que pode ser usado para tornar esta linha mais elegante: `discount.in?(1..100)`

Reply:
Eu diria que não tem mal em executar isto se o desconto for 0, simplesmente devia devolver 0 se estamos a falar de montante descontado
E nesse caso, podias fazer:
```rb
discount.to_i.clamp(0, 100)
```
que garante que mesmo nil é transformado em 0 para efeitos do cálculo, e podes então remover tudo o resto deixando só: 
```rb
def discount_value(discount)
  (price * (discount.clamp(0, 100) / 100)).round(2)
end
```